### PR TITLE
[WIP] Added Glock-17, Maxim-9 and Drum Magazines

### PR DESCRIPTION
**Pull request by Broken-Admin**
[What is a Maxim-9 pistol?](https://silencerco.com/maxim/)

### CODE REVIEW

**Comment by Cyrano7** [comment]
Having 4 versions of the same gun name is totally overkill and potentially confusing to the player. Is this baffle a separate item?  What use is this extended functionality to the player? Not trying to be rude, just trying to get a justification.

File: data/json/items/gun/9mm.json [hunk 1]
```diff
@@ -402,5 +402,171 @@
...
+    "target": "internal_silence_pistol_2",
+    "msg": "You remove one of the baffels."
+    }
+},
+{
+  "id": "internal_silence_pistol_2",
+  "type": "GUN",
+  "name": "Maxim-9",
+  "skill": "pistol",
+  "description": "Made by Silencer Co.  It has 1 of the removable baffels taken off.",
```
Comment (Cyrano7) [inline data/json/items/gun/9mm.json]: I'm pretty sure it's baffle not baffel.

Comment (Broken-Admin) [reply data/json/items/gun/9mm.json]: I'll fix that.

**Comment by Broken-Admin** [comment]
The justification is it's got 4 removable baffles to compromise silence for less weight and volume, the reason it has 4 versions is for it to be possible to change out baffles without C++.

**Comment by Broken-Admin** [comment]
Apparently I didn't commit that part to the repo.

**Comment by Cyrano7** [comment]
Just fyi 6 volume for the full version is bigger then an MP5.

**Comment by Broken-Admin** [comment]
The naming acted weird in reloading, you couldn't tell what was what. So I fix that.

**Comment by Broken-Admin** [comment]
@Cyrano7 How's it now?

**Comment by kevingranade** [comment]
I agree with Cyrano, 4 versions is pretty overkill.  I very much doubt anyone would use any configurations other than quietest and loudest.

Can you link a source (thats not a video) that describes how this works? I cant find any details on the silencerco website.

More importantly, if it physically adds and removes things from the gun, it needs to produce a baffle item when removed and require a baffle item when it's added.  You can define a new mod slot, something like, "maxim_baffle" and a maxim-specif suppressor item that occupies that slot.
Having the baffles pop in and out of existence is not ok.

**Comment by Broken-Admin** [comment]
I'm now planning to add it as mods.

**Comment by Cyrano7** [comment]
@Broken-Admin I'm going to close this for now since you stated that you want to make it a mod instead. Let me know if you change your mind.

**Comment by Broken-Admin** [comment]
Mod, as in *Gun Mod*

**Comment by Broken-Admin** [comment]
I wanted to PR the Glock-17 in a different PR, but I haven't been able to communicate with KG.

**Comment by Broken-Admin** [comment]
Please reopen btw...

**Comment by Broken-Admin** [comment]
Do we want Glock stuff to be added as separate PR?

**Comment by kevingranade** [comment]
Break it into as many different pieces as you want.  Making a PR too small
is never a problem if it adds functionality.

**Comment by Cyrano7** [comment]
@Broken-Admin Having smaller PRs or PRs that add or fix only one thing tend to get reviewed and merged faster under the current system from my experience. Sometimes features B and C need excessive rework or review and it can prevent people from looking a feature A.

**Comment by Broken-Admin** [comment]
Going to close this and only PR the Glock-17 and drum mags.

**Pull request closed without merging**